### PR TITLE
Groot now written in arbor

### PR DESCRIPTION
**Pull request by narendasan**
This branch represents groot written in the arbor framework. The system currently compiles and runs but testing is necessary. Hence the WIP label.

Note on merge this repo should be renamed to groot-api-gateway

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+	"github.com/acm-uiuc/arbor/server"
+	"github.com/acm-uiuc/groot-api-gateway/config"
```
Comment (aashishkapur) [inline server.go]: What's the difference between groot-api-gateway and arbor? They both map to the arbor repo

Comment (narendasan) [inline server.go]: so arbor is now a package you import. it contains 4 parts, the proxy, security, server and service and basically manage everything about the gateway. groot-api-gateway is the name of the api gateway app. and it just contains the routes and settings we want.

Comment (narendasan) [inline server.go]: So when this is merged there will be a slightly different process in how you install groot. you still symlink the api gateway to [gopath]/github.com/acm-uiuc/groot-api-gateway, but now you can just to a go install github.com/acm-uiuc/arbor/* to get all the packages

File: services/auth.go [hunk 2]
```diff
@@ -27,40 +28,40 @@ const AuthToken string = config.AuthPrefix + config.AuthToken
...
+	services.Route{
 		"EndSession",
 		"DELETE",
 		"/session/{token}",
 		EndSession,
 	},
 }
 
-//Route handler
+// services.Route handler
```
Comment (sameetandpotatoes) [inline services/auth.go]: This probably doesn't need to be renamed - same as in other files.

Comment (narendasan) [inline services/auth.go]: It does because it refers to arbor/services that defines the routes struct

Comment (narendasan) [inline services/auth.go]: if you are referring to the comment, yeah, was a find and replace artifact

**Comment by sameetandpotatoes** [comment]
Also, how has this been tested so far? Have you just run it locally and tested routing to each service?

File: services/auth.go (hunk 2 above)
Comment (sameetandpotatoes) [inline services/auth.go]: Yeah I was referring to the comment, sorry haha.

**Comment by narendasan** [comment]
Yeah its been fully with auth, groups. and some what with the other first gen services (mainly user functionality), and a little of the 2nd gen ones (since I dont have them fully setup)

File: build.sh [hunk 3]
```diff
@@ -13,24 +13,25 @@ mkdir -p log/prod
...
-go install github.com/acm-uiuc/groot/services
+go install github.com/acm-uiuc/groot-api-gateway/services
 
-echo Building groot
+echo Building groot-api-gateway
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 mkdir -p build 
 echo Placing binary in [PATH TO GROOT]/build
-(cd $DIR; go build -o  $DIR/build/groot ./server) 
+(cd $DIR; go build -o  $DIR/build/groot .)
```
Comment (bcongdon) [inline build.sh]: this should be changed so the binary output is named `groot-api-gateway`

**Comment by narendasan** [comment]
Anymore changes needed?

**Comment by aashishkapur** [comment]
The README install instructions need to be changed from `groot` to `groot-api-gateway`, under the First Time Setup instructions.  Besides that, I think everything looks good.

**Comment by narendasan** [comment]
Yeah will change documentation in another pr

**Comment by narendasan** [comment]
since arbor will be taking most of the documentation that is currently here

**Comment by bcongdon** [comment]
Yeah, I'd like to see this get merged sooner rather than later. Looks good to me.

**Comment by sameetandpotatoes** [comment]
Same 👍

**Comment by narendasan** [comment]
ok will approve and merge

**Comment by narendasan** [comment]
i guess I am not allowed to approve my own pr

**Pull request merged**